tests/test_orientation.py

```diff
@@ -72,8 +82,13 @@ def approx_ang(expected_degrees):
     def assert_angle(angle_radians, expected_degrees, message):
         expected_degrees = approx_ang(math.degrees(expected_degrees))
         # Check both +0 and +360 so approx can cover the jump between -180 and 180
-        assert math.degrees(angle_radians) == expected_degrees or math.degrees(angle_radians) + 360 == expected_degrees, \
-            message
+        if allow_wrapping:
+            assert \
+                math.degrees(angle_radians) == expected_degrees or \
+                math.degrees(angle_radians) + 360 == expected_degrees, \
+                message
```

Comment:
Why do we need to do this?

Reply:
so we can allow for a small error on a value which is close to 180 or -180. it means it checks both the 180 version and the -180 version in one check.

Reply:
Right, but these are tests -- they're dealing with fixed inputs which should mean that they have fixed outputs. We should know which side the result is and just assert that.

Are you saying that we get varying results from `solvePnP` even for constant input data?

An alternative way of spelling this, _iff_ it's absolutely needed, might be to declare that the tests operate in a positive-only frame and run all the values through `% 360` first.
I'm still not sure I like this approach as it means we're not really checking the direct output from the API. The not-checking-the-real-value aspect could be mitigated by adding a range bounds first, perhaps like this:
``` python
assert -180 < actual_degrees < 180, \
    "{} is outside the expected range (-180° to 180°)".format(angle_name)
    # angle_name is one of rot_x, rot_y, etc.

# Bound to positive numbers for easier comparisons
actual_degrees %= 360

assert actual_degrees == expected_degrees, ...
```

While I'll admit there's a bit more processing here, degrees being a modulo thing is fairly obvious to everyone since we're used to the idea that 720° is the direction same as 0°.

Reply:
I respect what you're saying but I disagree that we should check exact values. It makes the tests extremely fragile for changes higher up the pipeline (i.e. thresholding). If we want to check that the outputs don't vary, we should have a separate test

Reply:
`%360 doesn't solve the problem, it shifts the barrier to be between 0 and 360